nvflare/fuel/utils/config.py

```diff
@@ -0,0 +1,192 @@
+
+    @classmethod
+    def _config_format_extensions(cls):
+        return {
+            "JSON": [".json", ".json.default"],
+            "PYHOCON": [".conf", ".conf.default"],
+            "OMEGACONF": [".yml", ".yml.default"],
+        }
+
+    @classmethod
+    def search_order(cls):
+        return [cls.JSON, cls.PYHOCON, cls.OMEGACONF]
+
+    @classmethod
+    def get_extensions(cls, fmt: str) -> Optional[List[str]]:
+        return cls._config_format_extensions().get(fmt, None)
+
+    @classmethod
+    def config_exts(cls) -> str:
+        search_sequence: List[ConfigFormat] = cls.search_order()
+        extensions = []
+        for fmt in search_sequence:
+            exts = cls.get_extensions(fmt.name)
+            if exts:
+                extensions.extend(exts)
+
+        return "|".join(iter(extensions))
+
+
+class Config(ABC):
+    @abstractmethod
+    def get_native_conf(self):
+        """
+        Return the original underline config object representation if you prefer to use it directly
+        Pyhocon → ConfigTree
+        JSON → Dict
+        OMEGACONF → ConfigDict
+
+        Returns: Any, native config objects
+        """
+
+        pass
+
+    @abstractmethod
+    def get_format(self):
+        """
+        Returns: ConfigFormat, returns the current config objects ConfigFormat
+        """
+
+        pass
+
+    @abstractmethod
+    def get_location(self) -> Optional[str]:
+        """
+        Returns: None if the config is not from file else return the file path where this configuration is loaded from
+
+        """
+        pass
+
+    @abstractmethod
+    def to_dict(self) -> Dict:
+        """
+        convert underline config object to dictionary
+        Returns:Dict
+
+        """
+        pass
+
+    @abstractmethod
+    def to_conf(self, element: Dict) -> str:
+        """
+            convert dict element to the str representation of the underline configuration.
+            For example, for JsonFormat, the method return json string
+            for PyhoconFormat, the method return pyhocon string
+            for OmegaconfFormat, the method returns YAML string representation
+        Args:
+            element: dict
+
+        Returns: str string representation of the configuration in given format
+
+        """
+
+        pass
+
+    @abstractmethod
+    def get_int(self, key: str, default=None) -> Optional[int]:
+        pass
+
+    @abstractmethod
+    def get_float(self, key: str, default=None) -> Optional[float]:
+        pass
+
+    @abstractmethod
+    def get_bool(self, key: str, default=None) -> Optional[bool]:
+        pass
+
+    @abstractmethod
+    def get_str(self, key: str, default=None) -> Optional[str]:
+        pass
+
+    @abstractmethod
+    def get_list(self, key: str, default=None) -> Optional[List]:
+        pass
+
+    @abstractmethod
+    def get_config(self, key: str, default=None):
+        """
+           This method returns nvflare.fuel.utils.Config object
```

Comment:
Not sure I get this. What is nvflare.fuel.utils.Config object? Isn't it the same as the "self"?

Reply:
But I am not sure this is needed. If every loader simply returns a dict, the nested handling is only on dict. There is a dict_utils already. We could add more funcs there if needed. We don't need every Config type to do it.